use Vᴴ instead of Vh where appropriate in SVD docs

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61038
* **#61037**

Differential Revision: [D29491985](https://our.internmc.facebook.com/intern/diff/D29491985)

Commit:
warn about SVD outputs not supporting backprop

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29486485).

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29486485).

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491364).

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491985).

albanD commented on torch/linalg/__init__.py:
```diff
@@ -1492,6 +1492,8 @@
              The gradient will also be numerically unstable when :attr:`A` has small singular
              values, as it also depends on the computaiton of :math:`\frac{1}{\sigma_i}`.
 
+.. warning:: Backpropagation is only supported for input matrices with at most one zero eigenvalue.
```
I feel like the warning above actually mentions most of this already. Sorry for missing that.
We should fix the formatting of `Vh` though to be `V` or `V^H` with math formatting.

dagitses replied:
it looks like a similar warning exists in both variants of the function. So revert changes to both files and edit this file to use V^H instead of Vh? Should I do that across the board for this function or just the nearest warning?

dagitses replied:
I see some place already using Vᴴ. How does this look?

Commit:
Update on "warn about SVD outputs not supporting backprop"


Differential Revision: [D29491985](https://our.internmc.facebook.com/intern/diff/D29491985)

[ghstack-poisoned]

albanD reviewed (approved):
LGTM!

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491985).

Commit:
Update on "use Vᴴ instead of Vh where appropriate in SVD docs"


Differential Revision: [D29491985](https://our.internmc.facebook.com/intern/diff/D29491985)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491985).

Commit:
Update on "use Vᴴ instead of Vh where appropriate in SVD docs"


Differential Revision: [D29491985](https://our.internmc.facebook.com/intern/diff/D29491985)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491985).

Commit:
Update on "use Vᴴ instead of Vh where appropriate in SVD docs"


Differential Revision: [D29491985](https://our.internmc.facebook.com/intern/diff/D29491985)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29491985).

IvanYashchuk commented:
Note that when discussed improving linear algebra module's documentation it was proposed to use Unicode characters for mathematics, but we decided that LaTeX should be used instead. See https://github.com/pytorch/pytorch/issues/54878#issuecomment-821962806 and discussion in that issue.

Here in `torch.linalg.svd` `Vh` was used on purpose, it's the name of a returned variable, even though `Vᴴ` is a valid name for a Python variable, it could as well be named differently. NumPy uses `vh`, SciPy uses `Vh`. It can be confusing what the function actually returns when mixing mathematical expressions with the return signature.

cc: @Lezcano

Lezcano commented:
As @IvanYashchuk mentioned, there's both the discussion about not using unicode symbols in the docs due to problems in the rendering.

Now, the changes in this PR are not correct, as `Vh` refers to the name of a returned variable:
```python
print(torch.linalg.svd(torch.rand(3, 4)).Vh)
```

This PR should be reverted cc @albanD
